server/autolinkplugin/plugin.go

```diff
@@ -130,25 +150,18 @@ func (p *Plugin) ProcessPost(c *plugin.Context, post *model.Post) (*model.Post,
 		if origText != "" {
 			newText := origText
 
-			channel, cErr := p.API.GetChannel(post.ChannelId)
-			if cErr != nil {
-				p.API.LogError("Failed to get Channel", "error", cErr.Error())
-				return false
-			}
-			teamName := ""
-			if channel.TeamId != "" {
-				team, tErr := p.API.GetTeam(channel.TeamId)
-				if tErr != nil {
-					p.API.LogError("Failed to get Team", "error", tErr.Error())
-					return false
-				}
-				teamName = team.Name
-			}
-
 			for _, l := range conf.Links {
 				if len(l.Scope) == 0 {
 					newText = l.Replace(newText)
-				} else if teamName != "" && contains(teamName, channel.Name, l.Scope) {
+					continue
+				}
+
+				teamName, channelName, rsErr := p.resolveScope(post.ChannelId)
```

Comment:
1/5 suggestion: replace `resolveScope(channelId string) (string, string, *model.AppError)` with `inScope(scope, channelId string) bool`; then the loop becomes even simpler:
```go
	for _, l := range conf.Links {
		if p.inScope(l.Scope, post.ChannelId) {
			newText = l.Replace(newText)
		}
	}
```

Reply:
Sounds logical but, I’m suggesting that it doesn’t have to be the resolve function.
4/5 a new `inScope` function is created that calls resolve and does the rest of the logic